src/dune_config/config.mli

```diff
@@ -27,6 +27,9 @@ val global_lock : Toggle.t t
     reduces concurrency *)
 val cutoffs_that_reduce_concurrency_in_watch_mode : Toggle.t t
 
+(** whether dune should optimize file copying on Linux/MacOS *)
+val copy_file : [ `Portable | `Fast ] t
```

Comment:
I wonder if "fast" is a good choice in the long term. I can imagine that in future we can come up with several faster than portable implementations, neither of which could be clearly preferable in all circumstances. In that world, it would be awkward to choose between "fast" and something else. Choosing between "sendfile" and something else seems cleaner.

Not a very strong opinion but it's going to be hard to change the name in future, so we might as well discuss it now.

Reply:
`Sendfile` is a bit of a misnomer because this option controls the optimization on macos as well (which doesn't use sendfile). I suppose we could call the option `Sendfile_or_clonefile`, but that would be a bit ugly as well. I guess I can choose `Best` for now.

If we need more granular options, we can always add them. It's only removing existing options that would be painful.

Reply:
> `Sendfile` is a bit of a misnomer because this option controls the optimization on macos as well (which doesn't use sendfile).

You could allow both `Sendfile` and `Clonefile` I suppose. This could even be a list of preferences rather than a single value, so the user could say something like "try Clonefile first, and then fallback to portable if Clonefile isn't supported".

Reply:
Qualitative descriptions like "fast" or "best" seem nice but they often hide important details that make it hard to diagnose problems.